File: python/distortion_correct_aksk_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.utils import encode_to_base64
from ais_sdk.utils import decode_to_wave_file
from ais_sdk.distortion_correct import distortion_correct_aksk
from ais_sdk.utils import init_global_env
import json

if __name__ == '__main__':
    # Services currently support North China-Beijing 1 (cn-north-1), Asia Pacific-Hong Kong (ap-southeast-1)
    init_global_env('cn-north-1')

    #
    # access moderation distortion correct post data by ak,sk
    #
    app_key = '*************'
    app_secret = '************'

    demo_data_url = 'https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/vat-invoice.jpg'

    #call interface use the url correction is true means do not correction
    result = distortion_correct_aksk(app_key, app_secret, "", demo_data_url, True)
    result_obj = json.loads(result)
    if result_obj['result']['data'] != '':
        decode_to_wave_file(result_obj['result']['data'], 'data/modeation-distortion-aksk-1.png')
    else:
        print result

    # call interface use the file
    result = distortion_correct_aksk(app_key, app_secret, encode_to_base64('data/modeation-distortion.jpg'), '', True)
    if result_obj['result']['data'] != '':
        decode_to_wave_file(result_obj['result']['data'], 'data/modeation-distortion-aksk-2.png')
    else:
        print result

File: javascript/distortion_correct_aksk_demo.js
/**
 * 扭曲校正服务ak,sk方式请求的使用示例
 */
var discor = require("./ais_sdk/distortion_correct");
var utils = require("./ais_sdk/utils");

// 初始化服务的区域信息，目前支持华北-北京一(cn-north-1)、亚太-香港(ap-southeast-1)等区域信息
utils.initRegion(region="cn-north-1");

var app_key = "*************";
var app_secret = "************";

var filepath = "./data/modeation_distortion.jpg";
var data = utils.changeFileToBase64(filepath);

// obs链接需要和region区域一致，不同的region的obs资源不共享
demo_data_url = "https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/vat-invoice.jpg";

discor.distortion_correct_aksk(app_key, app_secret, data, "", true, function (result) {
    var resultObj = JSON.parse(result);

    if (resultObj.result.data !== "") {
        utils.getFileByBase64Str("./data/modeation_distortion-aksk-1.jpg", resultObj.result.data);
    }else{
        console.log(result);
    }
});

discor.distortion_correct_aksk(app_key, app_secret, "", demo_data_url, true, function (result) {
    var resultObj = JSON.parse(result);

    if (resultObj.result.data !== "") {
        utils.getFileByBase64Str("./data/modeation_distortion-aksk-2.jpg", resultObj.result.data);
    }else{
        console.log(result);
    }
});